src/plugins/data_source/server/plugin.ts

```diff
@@ -3,19 +3,31 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { PluginInitializerContext, CoreSetup, CoreStart, Plugin, Logger } from 'src/core/server';
+import {
+  PluginInitializerContext,
+  CoreSetup,
+  CoreStart,
+  Plugin,
+  Logger,
+  IContextProvider,
+  RequestHandler,
+} from '../../../../src/core/server';
+import { DataSourceClient } from './client/data_source_client';
+import { DataSourceRouteHandlerContext } from './data_source_route_handler_context';
 import { dataSource, credential } from './saved_objects';
 
 import { DataSourcePluginSetup, DataSourcePluginStart } from './types';
 
 export class DataSourcePlugin implements Plugin<DataSourcePluginSetup, DataSourcePluginStart> {
   private readonly logger: Logger;
+  private readonly dataSourceClient: DataSourceClient;
 
   constructor(initializerContext: PluginInitializerContext) {
     this.logger = initializerContext.logger.get();
+    this.dataSourceClient = new DataSourceClient(this.logger);
   }
 
-  public setup(core: CoreSetup) {
+  public async setup(core: CoreSetup) {
```

Comment:
why add async here? seems we are just returning an async function.

Reply:
good catch, will fix that